exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/metric_exporter/__init__.py

```diff
@@ -166,37 +158,22 @@ def __init__(
             }
         instrument_class_temporality.update(preferred_temporality or {})
 
+        OTLPExporterMixin.__init__(
+            self,
+            self._endpoint,
+            self._certificate_file,
+            self._headers,
+            self._timeout,
+            self._compression,
+            self._session,
+        )
+
         MetricExporter.__init__(
             self,
             preferred_temporality=instrument_class_temporality,
             preferred_aggregation=preferred_aggregation,
         )
 
-    def _export(self, serialized_data: str):
-        data = serialized_data
-        if self._compression == Compression.Gzip:
-            gzip_data = BytesIO()
-            with gzip.GzipFile(fileobj=gzip_data, mode="w") as gzip_stream:
-                gzip_stream.write(serialized_data)
-            data = gzip_data.getvalue()
-        elif self._compression == Compression.Deflate:
-            data = zlib.compress(bytes(serialized_data))
-
-        return self._session.post(
-            url=self._endpoint,
-            data=data,
-            verify=self._certificate_file,
-            timeout=self._timeout,
-        )
-
-    @staticmethod
-    def _retryable(resp: requests.Response) -> bool:
-        if resp.status_code == 408:
-            return True
-        if resp.status_code >= 500 and resp.status_code <= 599:
-            return True
-        return False
-
```

Comment:
It is not necessary to implement this suggestion in this PR, but I would also add the `_ProtobufEncoder` in the metrics exporter and reuse the export like the other two signals are doing.

Reply:
Let me know if you are working on this; otherwise, I will create a separate issue.

Reply:
Hey, I am working on this. Will push this in the current PR itself.

Reply:
Added this in the current PR.